### PR TITLE
update cruft and pydantic

### PR DESCRIPTION
**Pull request by mr-kamran-ali**
# Changes
- cruft template and dependencies update

### CODE REVIEW

File: mex/common/models/base.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def _get_field_names_allowing_none(cls) -> list[str]:
         """Build a cached list of fields can be set to None."""
         field_names: list[str] = []
         for field_name, field_info in cls.get_all_fields().items():
-            validator = TypeAdapter(field_info.annotation)
+            validator = TypeAdapter(field_info.annotation)  # type: ignore[var-annotated]
```
Comment (erichesse) [inline mex/common/models/base.py:128]:
```suggestion
            validator: TypeAdapter[Any] = TypeAdapter(field_info.annotation)
```

**Review by erichesse — COMMENTED** [review]
Looks good. Just one small issue, if mypy asks for type annotation we should try to provide them as much as possible

File: mex/common/models/base.py (hunk 1 above)
Comment (mr-kamran-ali) [reply mex/common/models/base.py]: suggestion applied: 8a0745dba3d3a2d25c3d74053a7c7a81db2776c6

**Review by erichesse — APPROVED** [review]
merge away!

**Pull request merged**